pilot/pkg/bootstrap/server.go

```diff
@@ -589,10 +534,19 @@ func (s *Server) initDNSTLSListener(dns string, tlsOptions TLSOptions) error {
 	return nil
 }
 
-// initialize secureGRPCServer - using DNS certs
-func (s *Server) initSecureGrpcServer(port string, keepalive *istiokeepalive.Options, tlsOptions TLSOptions) error {
+// initialize secureGRPCServer.
+func (s *Server) initSecureGrpcServer(args *PilotArgs, port string) error {
+	if features.IstiodService.Get() == "" {
+		return nil
+	}
+
+	if args.TLSOptions.CaCertFile == "" && s.ca == nil {
```

Comment:
Do we still have any code path where this would happen ? We do generate certs at startup if they don't exist. We may have some flag combination where we don't generate - I guess
if we don't use k8s or don't have RBAC permission for secrets.

Can you add a TODO to sort this out - maybe we should make sure we always have at least a self-signed certificate, even in test mode.